migrations/2_deploy_contracts.js

```diff
@@ -2,12 +2,15 @@
 
 const MainframeToken = artifacts.require('MainframeToken')
 const MainframeStake = artifacts.require('MainframeStake')
+const MainframeTokenDistribution = artifacts.require('MainframeTokenDistribution')
 const MainframeEscrow = artifacts.require('MainframeEscrow')
 
 module.exports = (deployer, network) => {
   deployer.deploy(MainframeToken).then(() => {
     return deployer.deploy(MainframeEscrow, MainframeToken.address)
   }).then(() => {
     return deployer.deploy(MainframeStake, MainframeEscrow.address)
+  }).then(() => {
```

Comment:
I think this deployment could be done in parallel to the others as it has a dependency on `MainframeToken` but not the other contracts, ex:

```js
const deployEscrowAndStake = () => {
  return deployer.deploy(MainframeEscrow, MainframeToken.address).then(() => {
    return deployer.deploy(MainframeStake, MainframeEscrow.address)
  })
}

deployer.deploy(MainframeToken).then(() => {
  return Promise.all([
    deployEscrowAndStake(),
    deployer.deploy(MainframeTokenDistribution, MainframeToken.address),
  ])
})
```
This way the deployment of `MainframeTokenDistribution` only waits for the deployment of `MainframeToken`, not the other 2.